x-pack/test/api_integration/apis/ml/data_frame_analytics/stop_spaces.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { COMMON_REQUEST_HEADERS } from '../../../../functional/services/ml/common_api';
+import { USER } from '../../../../functional/services/ml/security_common';
+
+export default ({ getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+  const ml = getService('ml');
+  const spacesService = getService('spaces');
+  const supertest = getService('supertestWithoutAuth');
+
+  const jobIdSpace3 = 'ihp_od_space3';
+  const jobIdSpace4 = 'ihp_od_space4';
+  const idSpace3 = 'space3';
+  const idSpace4 = 'space4';
+
+  async function runStopRequest(jobId: string, space: string, expectedStatusCode: number) {
+    const { body } = await supertest
+      .post(`/s/${space}/api/ml/data_frame/analytics/${jobId}/_stop`)
+      .auth(
+        USER.ML_POWERUSER_ALL_SPACES,
+        ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER_ALL_SPACES)
+      )
+      .set(COMMON_REQUEST_HEADERS)
+      .expect(expectedStatusCode);
+
+    return body;
+  }
+
+  let space3JobDestIndex: string;
+  let space4JobDestIndex: string;
+
+  describe('POST data_frame/analytics/{analyticsId}/_stop with spaces', function () {
+    before(async () => {
+      await esArchiver.loadIfNeeded('ml/bm_classification');
+      await spacesService.create({ id: idSpace3, name: 'space_three', disabledFeatures: [] });
+      await spacesService.create({ id: jobIdSpace4, name: 'space_four', disabledFeatures: [] });
+      // job config with high training percent so it takes longer to run
+      const jobConfigSpace3 = ml.commonConfig.getDFABmClassificationJobConfig(jobIdSpace3);
+      await ml.api.createDataFrameAnalyticsJob(jobConfigSpace3, idSpace3);
+
+      const jobConfigSpace4 = ml.commonConfig.getDFABmClassificationJobConfig(jobIdSpace4);
+      await ml.api.createDataFrameAnalyticsJob(jobConfigSpace4, idSpace4);
+      await ml.testResources.setKibanaTimeZoneToUTC();
+
+      space3JobDestIndex = jobConfigSpace3.dest.index;
+      space4JobDestIndex = jobConfigSpace4.dest.index;
+    });
+
+    after(async () => {
+      await spacesService.delete(idSpace3);
+      await spacesService.delete(idSpace4);
+      await ml.api.deleteIndices(space3JobDestIndex);
+      await ml.api.deleteIndices(space4JobDestIndex);
+      await ml.api.cleanMlIndices();
+      await ml.testResources.cleanMLSavedObjects();
+    });
+
+    it('should stop job from same space', async () => {
+      const body = await runStopRequest(jobIdSpace3, idSpace3, 200);
+      expect(body).to.have.property('stopped', true);
```

Comment:
We should check on the API level that the job actually stopped (see my comment above).